finetune/saver.py

```diff
@@ -129,21 +142,44 @@ def load(self, path):
 
     def get_scaffold_init_fn(self):
         
-        def init_fn(scaffold, session):
+        def init_fn(scaffold, session, model_portion=None):
+            self.var_val = []
             if self.variables is not None:
                 variables_sv = self.variables
             else:
                 variables_sv = dict()
             all_vars = tf.global_variables()
-            self.var_val = []
-            for var in all_vars:
-                for saved_var_name, saved_var in itertools.chain(variables_sv.items(), self.fallback.items()):
-                    if saved_var_name == var.name:
-                        for func in self.variable_transforms:
-                            saved_var = func(var.name, saved_var)
-                        var.load(saved_var, session)
-                        break
-                            
+            is_dict = False
+            if model_portion != 'entire_model': #we must be loading in the case of two separate estimators
+                print("loading variables for")
+                print(model_portion)
+                is_dict = True
+                assert model_portion in ['featurizer','target'], "Must be using separate estimators if loading before graph creation"
+                trainables = variables_sv
+                if model_portion == 'featurizer':
+                    norm_variable_scopes = ['b:0', 'g:0']
+                    base = [v for v in trainables if 'target' not in v]
+                    trainables = [v for v in base if 'adapter' in v or v[-3:] in norm_variable_scopes]    
+                elif model_portion == 'target':
+                    trainables = [v for v in trainables if 'target' in v]
+                loaded_vars = [v for v in all_vars if v.name in trainables]
+                for var in loaded_vars:
+                    name = var.name
+                    for saved_var_name, saved_var in itertools.chain(variables_sv.items(), loaded_vars):
```

Comment:
Whilst this is being refactored we can change this double loop for a dictionary lookup.